sdktests/custom_matchers.go

```diff
@@ -85,15 +85,11 @@ func EventIsCustomEvent(
 	o := ldvalue.ObjectBuild()
 	o.Set("kind", ldvalue.String("custom"))
 	o.Set("key", ldvalue.String(eventKey))
-	if inlineUser {
-		o.Set("user", eventUser.AsValue())
-		o.Set("userKey", ldvalue.Null())
-	} else {
-		o.Set("user", ldvalue.Null())
-		o.Set("userKey", ldvalue.String(eventUser.GetKey()))
-	}
-	o.Set("data", data)
-	o.Set("metricValue", ldvalue.CopyArbitraryValue(metricValue))
+	setPropertyConditionally(o, inlineUser, "user", eventUser.AsValue())
+	setPropertyConditionally(o, !inlineUser, "userKey", ldvalue.String(eventUser.GetKey()))
+	setPropertyConditionally(o, eventUser.IsAnonymous(), "contextKind", ldvalue.String("anonymousUser"))
```

Comment:
This piece, contextKind, is something we weren't testing before. The rest of the changes in this method are just streamlining how we handle nulls in the expected data.